lib/api/objectCopy.js

```diff
@@ -409,9 +407,14 @@ function objectCopy(authInfo, request, sourceBucket,
                     // object with same name, so long as the source is not
                     // the same as the destination
                     if (!sourceIsDestination && dataToDelete) {
-                        data.batchDelete(dataToDelete, request.method, null,
-                                logger.newRequestLoggerFromSerializedUids(
-                                    log.getSerializedUids()));
+                        const newDataStoreName =
+                            Array.isArray(destDataGetInfoArr) &&
```

Comment:
don't we have a cleaner way to access the destination data store name?

Reply:
it looks like we make sure destDataGetInfoArray is an array (https://github.com/scality/S3/pull/957/commits/0cc254142d1b3fe833211dd379eef6d6bd31d819#diff-5ce9bf7b97556522975c2fae1a6944efL283) so I guess we don't need all the checks.

Reply:
i meant, don't we have separate metadata of the new data store so we don't have to dig into the location array?

Reply:
we store the metadata just before this batch delete step, but we pass the same `destDataGetInfoArr`as the object location to store and `services.metadataStoreObject` doesn't return the object MD, so we'd have to make a separate call to get the new object MD to get the location directly from the metadata (unless we want to refactor)

Reply:
I do see there is a separate `dataStoreName` property for object MD but I don't think we set it for the new metadata in objectCopy...

Reply:
we should be. not even in @nicolas2bert's new pr?

Reply:
I think Nicolas introduces a change in his PR where he gets the dest location constraint which we might be able to use to set the `dataStoreName` prop appropriately. So far we're only copying an object to a destination location based on the request/source objMd location header (depending on whether we use REPLACE or COPY), I think.

Reply:
by so far you mean before Nicolas's pr? The three of us should discuss tomorrow.